lib/write.js

```diff
@@ -112,7 +112,9 @@ exports.usingBmap = (deviceFileDescriptor, options = {}) => {
  */
 exports.usingStreaming = (deviceFileDescriptor, options = {}) => {
   return new Bluebird((resolve, reject) => {
-    const checksumStream = new CRC32Stream();
+    const checksumStream = new ChecksumStream({
+      algorithms: options.checksumAlgorithms
```

Comment:
>  would it be worth skipping the ChecksumStream entirely?

Yup, would be nice, but I think we can do that later, especially since that's where I was going with the whole [pipage](https://github.com/resin-io-modules/pipage) stuff.
It is a bit of a micro-optimisation though, as our bottlenecks are in other places that are actually I/O bound, but nevertheless – the fewer streams and intermediate buffers we need, to make something happen, the better.